extensions/io/org.eclipse.smarthome.io.mqttembeddedbroker/src/main/java/org/eclipse/smarthome/io/mqttembeddedbroker/internal/EmbeddedBrokerServiceImpl.java

```diff
@@ -0,0 +1,304 @@
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.eclipse.smarthome.config.core.ConfigurableService;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.io.mqttembeddedbroker.Constants;
+import org.eclipse.smarthome.io.mqttembeddedbroker.internal.MqttEmbeddedBrokerDetectStart.MqttEmbeddedBrokerStartedListener;
+import org.eclipse.smarthome.io.mqttembeddedbroker.internal.MqttEmbeddedBrokerMetrics.BrokerMetricsListener;
+import org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection;
+import org.eclipse.smarthome.io.transport.mqtt.MqttConnectionObserver;
+import org.eclipse.smarthome.io.transport.mqtt.MqttConnectionState;
+import org.eclipse.smarthome.io.transport.mqtt.MqttService;
+import org.eclipse.smarthome.io.transport.mqtt.MqttServiceObserver;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.moquette.BrokerConstants;
+import io.moquette.server.Server;
+import io.moquette.server.config.MemoryConfig;
+import io.moquette.spi.security.IAuthorizator;
+
+/**
+ * The {@link EmbeddedBrokerServiceImpl} starts the embedded broker, creates a
+ * {@link MqttBrokerConnection} and adds it to the {@link MqttService}.
+ *
+ * TODO: wait for NetworkServerTls implementation to enable secure connections as well
+ *
+ * @author David Graeff - Initial contribution
+ */
+@Component(immediate = true, service = EmbeddedBrokerService.class, configurationPid = "org.eclipse.smarthome.mqttembeddedbroker", property = {
+        org.osgi.framework.Constants.SERVICE_PID + "=org.eclipse.smarthome.mqttembeddedbroker",
+        ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=mqtt:mqttembeddedbroker",
+        ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=MQTT",
+        ConfigurableService.SERVICE_PROPERTY_LABEL + "=MQTT Embedded Broker" })
+public class EmbeddedBrokerServiceImpl implements EmbeddedBrokerService, ConfigurableService, MqttConnectionObserver,
+        MqttServiceObserver, MqttEmbeddedBrokerStartedListener, BrokerMetricsListener {
+    private MqttService service;
+    // private NetworkServerTls networkServerTls; //TODO wait for NetworkServerTls implementation
+
+    protected Server server;
+    private final Logger logger = LoggerFactory.getLogger(EmbeddedBrokerServiceImpl.class);
+    protected MqttEmbeddedBrokerDetectStart detectStart = new MqttEmbeddedBrokerDetectStart(this);
+    protected MqttEmbeddedBrokerMetrics metrics = new MqttEmbeddedBrokerMetrics(this);
+
+    private MqttBrokerConnection connection;
+
+    @Reference(cardinality = ReferenceCardinality.MANDATORY)
```

Comment:
`cardinality = ReferenceCardinality.MANDATORY` is the default for `@Reference`, please remove